src/project/python.rs

```diff
@@ -0,0 +1,130 @@
+use crate::utils::spanned::PixiSpanned;
+use indexmap::IndexMap;
+use serde::de::{DeserializeSeed, Error, MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::collections::HashMap;
+use std::fmt::Formatter;
+
+/// Represents a set of python dependencies on which a project can depend. The dependencies are
+/// formatted using a modern version specifier. See [`pep508_rs::modern::RequirementModern`].
+#[derive(Default, Debug, Clone)]
+pub struct PythonDependencies {
+    requirements: IndexMap<
+        PixiSpanned<rip::PackageName>,
+        PixiSpanned<(pep508_rs::modern::RequirementModern, pep508_rs::Requirement)>,
+    >,
+}
+
+impl PythonDependencies {
+    /// Returns `true` if no requirements have been specified
+    pub fn is_empty(&self) -> bool {
+        self.requirements.is_empty()
+    }
+
+    /// Returns the requirements as [`pep508_rs::Requirement`]s.
+    pub fn as_pep508(&self) -> Vec<pep508_rs::Requirement> {
```

Comment:
Looking at https://peps.python.org/pep-0508/ `"*"` should be a valid version but i get the following error:
```
 × failed to parse pixi.toml from /home/rarts/testpy
  ╰─▶ failed to parse project manifest
    ╭─[pixi.toml:15:1]
 15 │ [python-dependencies]
 16 │ pytest = "*"
    ·          ─┬─
    ·           ╰── data did not match any variant of untagged enum RequirementModern
    ╰────
    ```

Reply:
Lets also do in another PR.

Reply:
See https://github.com/prefix-dev/pixi/issues/466